src/Common/Editor/Block_Logic.php

```diff
@@ -0,0 +1,203 @@
+		}
+
+		return $this->return_should_load_blocks( true, 'block_editor' );
+	}
+
+	/**
+	 * Return the value of should_load_blocks passed through a filter.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool   $should_load The value of should_load_blocks.
+	 * @param string $reason      The reason why we are returning this value.
+	 *
+	 * @return bool
+	 */
+	private function return_should_load_blocks( bool $should_load, string $reason ): bool {
+		/**
+		 * Filter to determine if blocks should be loaded.
+		 *
+		 * This filter is used to determine if blocks should be loaded in the editor. The
+		 * $reason parameter can be used to determine why we are returning this value.
+		 *
+		 * @see self::should_load_blocks() for the reasons.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool       $should_load Whether blocks should be loaded.
+		 * @param string     $reason      The reason why we are returning this value.
+		 * @param ?WP_Screen $screen      The current screen object, if it has been set.
+		 */
+		return (bool) apply_filters( 'tec_common_should_load_blocks', $should_load, $reason, $this->screen );
+	}
+
+	/**
+	 * Determine if blocks should be loaded for a specific post type.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $post_type The post type to check.
+	 *
+	 * @return bool
+	 */
+	public function should_load_blocks_for_post_type( string $post_type = '' ): bool {
+		// Set up a static variable to ensure we don't check the same post type multiple times.
+		static $checked_post_types = [];
+
+		// If we weren't provided a post type, try to determine the current post type.
+		if ( empty( $post_type ) ) {
+			try {
+				$post_type = $this->determine_current_post_type();
+			} catch ( Exception $e ) {
+				return false;
+			}
+		}
+
+		// If we have already checked this post type, return the cached value.
+		if ( isset( $checked_post_types[ $post_type ] ) ) {
+			return $checked_post_types[ $post_type ];
+		}
+
+		/**
+		 * Filter to determine what post types should have blocks loaded.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $load_blocks_post_types The post types that should have blocks loaded.
+		 */
+		$load_blocks_post_types = (array) apply_filters( 'tec_common_load_blocks_post_types', [] );
+
+		// Store the post type in the static variable to avoid checking it again.
+		$checked_post_types[ $post_type ] = in_array( $post_type, $load_blocks_post_types, true );
+
+		return $checked_post_types[ $post_type ];
+	}
+
+	/**
+	 * Determine the current post type.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The current post type.
+	 * @throws Exception If the current post type cannot be determined.
+	 */
+	private function determine_current_post_type(): string {
+		// If we have the screen object, use its post type.
+		if ( null !== $this->screen ) {
+			return $this->screen->post_type;
+		}
+
+		// Try to use the global post object.
+		if ( isset( $GLOBALS['post']->post_type ) ) {
```

Comment:
Will this work reliably in REST API and Ajax context?

Reply:
> Will this work reliably in REST API and Ajax context?

That's a good question: I think it would make sense to have a separate case for those situations outside of this method, since the post type isn't as relevant in those contexts.